tests/core/providers/test_auto_provider.py

```diff
@@ -32,9 +17,9 @@ def env(name, tempval):
         ('file:///root/path/to/file.ipc', IPCProvider, {'ipc_path': '/root/path/to/file.ipc'}),
     ),
 )
-def test_load_provider_from_env(uri, expected_type, expected_attrs):
-    with env('WEB3_PROVIDER_URI', uri):
-        provider = load_provider_from_environment()
-        assert isinstance(provider, expected_type)
-        for attr, val in expected_attrs.items():
-            assert getattr(provider, attr) == val
+def test_load_provider_from_env(monkeypatch, uri, expected_type, expected_attrs):
+    monkeypatch.setenv('WEB3_PROVIDER_URI', uri)
```

Comment:
I ❤️ this fixture so much.  Also look at `tmpdir` if you haven't already.

Reply:
Yeah, these are great. Such a concise way to set up the test environment.